conans/test/integration/lockfile/test_user_overrides.py

```diff
@@ -211,3 +214,105 @@ def test_lock_add_error():
     c = TestClient()
     c.run(f"lock add --requires=math/1.0:pid1", assert_error=True)
     assert "ERROR: Invalid recipe reference 'math/1.0:pid1' is a package reference" in c.out
+
+
+class TestLockRemove:
+    @pytest.mark.parametrize("args, removed", [
+        ("--requires=math/*", ["math"]),
+        ("--requires=math/2.0", []),
+        ("--build-requires=cmake/1.0", ["cmake"]),
+        # Not valid ("--build-requires=*", ["cmake", "ninja"]),
+        ("--build-requires=*/*", ["cmake", "ninja"]), # But this is valid
+        ("--python-requires=mytool/*", ["mytool"]),
+        ("--python-requires=*tool/*", ["mytool", "othertool"]),
+        # With version ranges
+        ('--requires="math/[>=1.0 <2]"', ["math"]),
+        ('--requires="math/[>1.0]"', []),
```

Comment:
I dont think it works, but what about `--requires="*/[>=1.0]"`?

Reply:
No, that doesn't work yet, the name has to match for version ranges, I thought it didn't make sense to remove certain range for all versions, but the truth is that the patterns without version ranges can do it, so I'll make it the same and accept patterns too for the name.

Reply:
Done, please check